src/scala/microsoft-spark-2-4/src/main/scala/org/apache/spark/deploy/dotnet/DotNetUserAppException.scala

```diff
@@ -0,0 +1,15 @@
+package org.apache.spark.deploy.dotnet
+
+import org.apache.spark.SparkException
+
+/**
+ * This exception type describes an exception thrown by a .NET user application.
+ *
+ * @param exitCode Exit code returned by the .NET application.
+ * @param dotNetStackTrace Stacktrace extracted from .NET application logs.
+ */
+private[spark] class DotNetUserAppException(exitCode: Int, dotNetStackTrace: Option[String])
+    extends SparkException(dotNetStackTrace match {
```

Comment:
I assume there's no way to extend SparkUserAppException ?

Reply:
Yes, SparkUserAppException is private and has no visibility here

Reply:
SparkUserAppException is `private[spark]` and so DotNetUserAppException should have access to the class since the current package is defined as `org.apache.spark`.deploy.dotnet 

I suspect there may be issues due to SparkUserAppException 's limited constructors.